### DISCUSSION TITLE
Shortcode for embedding a Quarto RevealJS presentation inside of a website

### DISCUSSION THREAD

**Discussion by coatless** [quarto-dev/quarto-cli#7410] [Feature Requests]
### Description

From the [`revealjs` format's](https://quarto.org/docs/presentations/revealjs/) documentation page, there is way to take a RevealJS presentation and embedded it within a Quarto website. Specifically, the [source code](https://github.com/quarto-dev/quarto-web/blob/c7c98dd771f0d718aea615603708d0a7bb8b0cf3/docs/presentations/revealjs/index.qmd#L10-L16) uses the following `iframe` "trick": 

````html
<div>
```{=html}
<iframe class="slide-deck" src="demo/"></iframe>
```
</div>
````

Would it be possible to get an official shortcode to embed a Quarto presentation inside of a HTML Document/Website/Book? Something like: 

```
{{< presentation dname="example" >}}
```

Or is this something that should be placed for the moment in an extension?

**Comment by coatless** [comment]
I wrote an extension called `embedio` that handles embedding `revealjs` documents alongside of a few other formats. 

Demo: 

<img width="1454" alt="Screenshot 2024-04-22 at 12 49 20 PM" src="https://github.com/quarto-dev/quarto-cli/assets/833642/cb454b4e-c8af-4a66-8fac-563c7b4f9595">

You can check it out here:

https://quarto.thecoatlessprofessor.com/embedio/qembedio-embed-revealjs.html

Repository:

https://github.com/coatless-quarto/embedio

**Reply by mcanouil** [reply]
I added it to https://m.canouil.dev/quarto-extensions/extensions.html

**Reply by coatless** [reply]
@mcanouil thanks! how's this different than https://quarto.org/docs/extensions/listing-filters.html

Should I have this on my radar when adding more extensions?

**Reply by mcanouil** [reply]
It's a bit more comprehensive, comes from the Awesome Quarto initially (which was there a bit before the extension listings).

The listings I shared works using GitHub API to retrieve topics (used as categories) and description (you don't need to make a PR, to update something, unless you change the repository name hosting the extension). It already includes 141 extensions.
It's the first version, I intend to improve the whole thing over (my spare) time.

In both cases, those are only listings, not repository/archive for extensions.
